lua_cjson.c

```diff
@@ -1441,6 +1441,10 @@ static int json_decode(lua_State *l)
     if (token.type != T_END)
         json_throw_parse_error(l, &json, "the end", &token);
 
+    /* Make sure T_END (\x00) doesn't occur at the middle of input */
+    if (json.data + json_len > json.ptr)
+        json_throw_parse_error(l, &json, "EOF", &token);
```

Comment:
Is making this error handling become an instance configuration a good idea? 

Most JSON libraries from various languages can fail with an error in this case, and I believe this is a bug. However, it's sitting there for a very long time(in Kong Gateway)... : ( 

This PR is going to fix it entirely like a nuclear bomb. All `cjson.decode` in Kong Gateway will be affected, even the code from custom plugins and custom patches.